### ISSUE TITLE
OOVPA for CreateVertexShader_1_0_3925 defined twice

### ISSUE THREAD

**Issue by PatrickvL** [Cxbx-Reloaded/Cxbx-Reloaded#111]
These two OOVPA's : IDirect3DDevice8_CreateVertexShader_1_0_3925 and D3DDevice_CreateVertexShader_1_0_3925 should be merged into one (preferably using a verified source pattern/xbe).

After that, their duplicate mention in D3D8_1_0_3925 should be merged into one.

**Comment by PatrickvL** [comment]
Here a dump all Dxbx patterns for CreateVertexShader : 
(With the format : symbol, buildnr, library, crclen, crc, len, pattern, xrefs*)

_D3DDevice_CreateVertexShader@16 3911 d3d8 1F 29D0 0140 5153558B6C241485ED5674100FB745000FB74D028944240C894C241833F685ED........C1E0028D1C308D836C010000506A00E8........8BF085F6750C5E5DB80E0007805B59C210005733C0B95B0000008BFEF3AB8B4C2424894E04C706010000008D4630B9140000008D9B00000000C7000200000083C0104975F48B7C241885FF740F33D285ED0F94C2525756E8........33C085ED744D8B7E048B4C241083CF0481F978200000897E048BC7740683C80189460481F9787300007504834E04088B44241C894610C1E0025083C504558DBE6801000057E8........2BC78B7C2418C1F80285FF740E8D8C86680100005157E8........8B542420C1EB028BC3894614C78486680100000000000083CE015F89325E5D33C05B59C210009090 ^003FR ?ParseDeclarationConstants@D3D@@YGKPBKPAK@Z ^0053R _LocalAlloc@8 ^00AFR ?ParseDeclarationStream@D3D@@YGXPAUVertexShader@1@PBKH@Z ^00F9R ?ParseProgram@D3D@@YGPAKPAKPBKK@Z ^0114R ?ParseDeclarationConstants@D3D@@YGKPBKPAK@Z
_D3DDevice_CreateVertexShader@16 4361 d3d8 1F 29D0 0140 5153558B6C241485ED5674100FB745000FB74D028944240C894C241833F685ED........C1E0028D1C308D8318010000506A00E8........8BF085F6750C5E5DB80E0007805B59C210005733C0B9460000008BFEF3AB8B4C2424894E04C706010000008D461CB9100000008D9B00000000C7000200000083C0104975F48B7C241885FF740F33D285ED0F94C2525756E8........33C085ED744D8B7E048B4C241083CF1081F978200000897E048BC7740683C80189460481F9787300007504834E04088B44241C894608C1E0025083C504558DBE1401000057E8........2BC78B7C2418C1F80285FF740E8D8C86140100005157E8........8B542420C1EB028BC389460CC78486140100000000000083CE015F89325E5D33C05B59C210009090 ^003FR ?ParseDeclarationConstants@D3D@@YGKPBKPCK@Z ^0053R _LocalAlloc@8 ^00AFR ?ParseDeclarationStream@D3D@@YGXPAUVertexShader@1@PBKH@Z ^00F9R ?ParseProgram@D3D@@YGPCKPCKPBKK@Z ^0114R ?ParseDeclarationConstants@D3D@@YGKPBKPCK@Z
_D3DDevice_CreateVertexShader@16 4627 d3d8 1F 29D0 0140 5153558B6C241485ED5674100FB745000FB74D028944240C894C241833F685ED........C1E0028D1C308D8318010000506A00E8........8BF085F6750C5E5DB80E0007805B59C210005733C0B9460000008BFEF3AB8B4C2424894E04C706010000008D461CB9100000008D9B00000000C7000200000083C0104975F48B7C241885FF740F33D285ED0F94C2525756E8........33C085ED744D8B7E048B4C241083CF1081F978200000897E048BC7740683C80189460481F9787300007504834E04088B44241C894608C1E0025083C504558DBE1401000057E8........2BC78B7C2418C1F80285FF740E8D8C86140100005157E8........8B542420C1EB028BC389460CC78486140100000000000083CE015F89325E5D33C05B59C210009090 ^003FR ?ParseDeclarationConstants@D3D@@YGKPBKPCK@Z ^0053R _LocalAlloc@8 ^00AFR ?ParseDeclarationStream@D3D@@YGXPAUVertexShader@1@PBKH@Z ^00F9R ?ParseProgram@D3D@@YGPCKPCKPBKK@Z ^0114R ?ParseDeclarationConstants@D3D@@YGKPBKPCK@Z
_D3DDevice_CreateVertexShader@16 5344 d3d8 1F 29D0 0140 5153558B6C241485ED5674100FB745000FB74D028944240C894C241833F685ED........C1E0028D1C3068000080248D831801000050E8........8BF085F6750C5E5DB80E0007805B59C210005733C0B9460000008BFEF3AB8B4C2424894E04C706010000008D461CB9100000008D4900C7000200000083C0104975F48B7C241885FF740F33D285ED0F94C2525756E8........33C085ED744D8B7E048B4C241083CF1081F978200000897E048BC7740683C80189460481F9787300007504834E04088B44241C894608C1E0025083C504558DBE1401000057E8........2BC78B7C2418C1F80285FF740E8D8C86140100005157E8........8B542420C1EB028BC389460CC78486140100000000000083CE015F89325E5D33C05B59C210009090 ^003FR ?ParseDeclarationConstants@D3D@@YGKPBKPCK@Z ^0056R _XMemAlloc@8 ^00AFR ?ParseDeclarationStream@D3D@@YGXPAUVertexShader@1@PBKH@Z ^00F9R ?ParseProgram@D3D@@YGPCKPCKPBKK@Z ^0114R ?ParseDeclarationConstants@D3D@@YGKPBKPCK@Z
_D3DDevice_CreateVertexShader@16 5558 d3d8 1F 29D0 013E 5153558B6C241485ED5674100FB745000FB74D028944240C894C241833F685ED........C1E0028D1C3068000080248D831801000050E8........8BF085F6750C5E5DB80E0007805B59C210005733C0B9460000008BFEF3AB8B4C2424894E04C706010000008D461CB9100000008D4900C7000200000083C0104975F48B7C241885FF740F33D285ED0F94C2525756E8........33C085ED744D8B7E048B4C241083CF1081F978200000897E048BC7740683C80189460481F9787300007504834E04088B44241C894608C1E0025083C504558DBE1401000057E8........2BC78B7C2418C1F80285FF740E8D8C86140100005157E8........8B542420C1EB028BC389460CC78486140100000000000083CE015F89325E5D33C05B59C21000 ^003FR ?ParseDeclarationConstants@D3D@@YGKPBKPCK@Z ^0056R _XMemAlloc@8 ^00AFR ?ParseDeclarationStream@D3D@@YGXPAUVertexShader@1@PBKH@Z ^00F9R ?ParseProgram@D3D@@YGPCKPCKPBKK@Z ^0114R ?ParseDeclarationConstants@D3D@@YGKPBKPCK@Z
_D3DDevice_CreateVertexShader@16 5659 d3d8 1F 29D0 013E 5153558B6C241485ED5674100FB745000FB74D028944240C894C241833F685ED........C1E0028D1C3068000080248D831801000050E8........8BF085F6750C5E5DB80E0007805B59C210005733C0B9460000008BFEF3AB8B4C2424894E04C706010000008D461CB9100000008D4900C7000200000083C0104975F48B7C241885FF740F33D285ED0F94C2525756E8........33C085ED744D8B7E048B4C241083CF1081F978200000897E048BC7740683C80189460481F9787300007504834E04088B44241C894608C1E0025083C504558DBE1401000057E8........2BC78B7C2418C1F80285FF740E8D8C86140100005157E8........8B542420C1EB028BC389460CC78486140100000000000083CE015F89325E5D33C05B59C21000 ^003FR ?ParseDeclarationConstants@D3D@@YGKPBKPCK@Z ^0056R _XMemAlloc@8 ^00AFR ?ParseDeclarationStream@D3D@@YGXPAUVertexShader@1@PBKH@Z ^00F9R ?ParseProgram@D3D@@YGPCKPCKPBKK@Z ^0114R ?ParseDeclarationConstants@D3D@@YGKPBKPCK@Z
_D3DDevice_CreateVertexShader@16 5788 d3d8 1F 29D0 013E 5153558B6C241485ED5674100FB745000FB74D028944240C894C241833F685ED........C1E0028D1C3068000080248D831801000050E8........8BF085F6750C5E5DB80E0007805B59C210005733C0B9460000008BFEF3AB8B4C2424894E04C706010000008D461CB9100000008D4900C7000200000083C0104975F48B7C241885FF740F33D285ED0F94C2525756E8........33C085ED744D8B7E048B4C241083CF1081F978200000897E048BC7740683C80189460481F9787300007504834E04088B44241C894608C1E0025083C504558DBE1401000057E8........2BC78B7C2418C1F80285FF740E8D8C86140100005157E8........8B542420C1EB028BC389460CC78486140100000000000083CE015F89325E5D33C05B59C21000 ^003FR ?ParseDeclarationConstants@D3D@@YGKPBKPCK@Z ^0056R _XMemAlloc@8 ^00AFR ?ParseDeclarationStream@D3D@@YGXPAUVertexShader@1@PBKH@Z ^00F9R ?ParseProgram@D3D@@YGPCKPCKPBKK@Z ^0114R ?ParseDeclarationConstants@D3D@@YGKPBKPCK@Z
_D3DDevice_CreateVertexShader@16 5849 d3d8 1F 29D0 013E 5153558B6C241485ED5674100FB745000FB74D028944240C894C241833F685ED........C1E0028D1C3068000080248D831801000050E8........8BF085F6750C5E5DB80E0007805B59C210005733C0B9460000008BFEF3AB8B4C2424894E04C706010000008D461CB9100000008D4900C7000200000083C0104975F48B7C241885FF740F33D285ED0F94C2525756E8........33C085ED744D8B7E048B4C241083CF1081F978200000897E048BC7740683C80189460481F9787300007504834E04088B44241C894608C1E0025083C504558DBE1401000057E8........2BC78B7C2418C1F80285FF740E8D8C86140100005157E8........8B542420C1EB028BC389460CC78486140100000000000083CE015F89325E5D33C05B59C21000 ^003FR ?ParseDeclarationConstants@D3D@@YGKPBKPCK@Z ^0056R _XMemAlloc@8 ^00AFR ?ParseDeclarationStream@D3D@@YGXPAUVertexShader@1@PBKH@Z ^00F9R ?ParseProgram@D3D@@YGPCKPCKPBKK@Z ^0114R ?ParseDeclarationConstants@D3D@@YGKPBKPCK@Z
_D3DDevice_CreateVertexShader@16 5933 d3d8 1F 29D0 013E 5153558B6C241485ED5674100FB745000FB74D028944240C894C241833F685ED........C1E0028D1C3068000080248D831801000050E8........8BF085F6750C5E5DB80E0007805B59C210005733C0B9460000008BFEF3AB8B4C2424894E04C706010000008D461CB9100000008D4900C7000200000083C0104975F48B7C241885FF740F33D285ED0F94C2525756E8........33C085ED744D8B7E048B4C241083CF1081F978200000897E048BC7740683C80189460481F9787300007504834E04088B44241C894608C1E0025083C504558DBE1401000057E8........2BC78B7C2418C1F80285FF740E8D8C86140100005157E8........8B542420C1EB028BC389460CC78486140100000000000083CE015F89325E5D33C05B59C21000 ^003FR ?ParseDeclarationConstants@D3D@@YGKPBKPCK@Z ^0056R _XMemAlloc@8 ^00AFR ?ParseDeclarationStream@D3D@@YGXPAUVertexShader@1@PBKH@Z ^00F9R ?ParseProgram@D3D@@YGPCKPCKPBKK@Z ^0114R ?ParseDeclarationConstants@D3D@@YGKPBKPCK@Z

**Comment by LukeUsher** [comment]
Looking at those patterns, there are enough similarities that I am confident that a single OOVPA pattern can be written which will match all versions of this function

**Comment by PatrickvL** [comment]
I agree - here's a version that covers the first 41 bytes :
```
// ******************************************************************
// * IDirect3DDevice8_CreateVertexShader
// ******************************************************************
// Dxbx pattern for 3911, 4361, 4627, 5344, 5558, 5659, 5788, 5849, 5933:
// 51 53 55 8B 6C 24 14 85 ED 56 74 10 0F B7 45 00 0F B7 4D 02 89 44 24 0C 89 4C 24 18 33 F6 85 ED .. .. .. .. C1 E0 02 8D 1C 30
// 00 01 02 03 04 05 06 07 08 09 0A 0B 0C 0D 0E 0F 10 11 12 13 14 15 16 17 18 19 1A 1B 1C 1D 1E 1F 20 21 22 23 24 25 26 27 28 29
OOVPA_NO_XREF(IDirect3DDevice8_CreateVertexShader_1_0_3925, 38)
		// IDirect3DDevice8_CreateVertexShader+0x00 : push ecx; push ebx; push ebp
		{ 0x00, 0x51 },
		{ 0x01, 0x53 },
		{ 0x02, 0x55 },
		
		{ 0x03, 0x8B },
		{ 0x04, 0x6C },
		{ 0x05, 0x24 },
		{ 0x06, 0x14 },
		{ 0x07, 0x85 },
		{ 0x08, 0xED },
		{ 0x09, 0x56 },

		// IDirect3DDevice8_CreateVertexShader+0x0A : jz +0x10
		{ 0x0A, 0x74 },
		{ 0x0B, 0x10 },

		{ 0x0C, 0x0F },
		{ 0x0D, 0xB7 },
		{ 0x0E, 0x45 },
		{ 0x0F, 0x00 },
		{ 0x10, 0x0F },
		{ 0x11, 0xB7 },
		{ 0x12, 0x4D },
		{ 0x13, 0x02 },
		{ 0x14, 0x89 },
		{ 0x15, 0x44 },
		{ 0x16, 0x24 },
		{ 0x17, 0x0C },
		{ 0x18, 0x89 },
		{ 0x19, 0x4C },
		{ 0x1A, 0x24 },
		{ 0x1B, 0x18 },
		{ 0x1C, 0x33 },
		{ 0x1D, 0xF6 },
		{ 0x1E, 0x85 },
		{ 0x1F, 0xED },

		{ 0x24, 0xC1 },
		{ 0x25, 0xE0 },
		{ 0x26, 0x02 },
		{ 0x27, 0x8D },
		{ 0x28, 0x1C },
		{ 0x29, 0x30 },
OOVPA_END;
```

Haven't tested it though.

**Comment by LukeUsher** [comment]
I fixed the pattern, the last 6 offsets are incorrect, they should be the following:

`		{ 0x43, 0xC1 },
		{ 0x44, 0xE0 },
		{ 0x45, 0x02 },
		{ 0x46, 0x8D },
		{ 0x47, 0x1C },
		{ 0x48, 0x30 },`

I suspect there was an mistake in converting the .pat files to the patterns you listed above, the format of the  IDA compatible .PAT files is as follows:

<leading bytes> <CRC16 len> <CRC16> <total length> <public name(s)> <referenced name(s)> <tailing bytes>

The tailing bytes start at 0x20 + <CRC16 len>, the leading bytes always counts the first 32-bytes.

**Comment by PatrickvL** [comment]
Ah that explains it, thanks. I'll have to generate the "sorted reordered" version again. Values at offset 31 (0x1F) should be ignored until then.

**Comment by PatrickvL** [comment]
Now that the patterns are re-created, this was an easy fix - see PR #172
